### DISCUSSION TITLE
Install Damask in linux-aarch64 platform？

### DISCUSSION THREAD

**Discussion by 123fgh123** [damask-multiphysics/DAMASK#42] [Q&A]
Is it possible to install Damask in linux-aarch64 platform？I have not found the conda package of Damask for linux-aarch64.  https://anaconda.org/conda-forge/damask

As an alternative，is it possible to install Damask in linux-aarch64 by Source Code ?  

The Source Code installation procedure provided in the web (https://www.damask.mpie.de/installation/source_code.html) seems to be quite brief, especially for the installation of prerequisites such as PETSc. Is it poosible to provide a more detailed description to the installation procedure of prerequisites？

Thank you very much！

**Comment by ruanqc** [comment]
Hi, basically installation main steps are illustrated in the DAMASK web. 1. download and unpack damsk source code, 2. download and install petsc, 3. cmake to install solver, 4. install pre and post processing tools. In terms of the installation of PETSC, you need to download the PETSC package, and ./configure in the unzip petsc dir,

''./configure \
    --with-fc=$F90 --with-cc=$CC --with-cxx=$CXX \
    --download-fblaslapack \
    --download-hdf5 \
    --download-zlib \
    --download-fftw \
    --download-superlu \
    --download-hypre \
    --download-mumps \
    --download-ml \
    --download-mpich \
    --download-chaco \
    --download-metis\
    --download-parmetis \
    --download-scalapack \
    --download-suitesparse \
    --download-superlu_dist \
    --download-triangle \
    --with-cxx-dialect=C++11 \
    --with-c2html=0 \
    --with-debugging=0 \
    --with-ssl=0 \
    --with-x=0 \

   COPTFLAGS="-O3 -xHost -no-prec-div" CXXOPTFLAGS="-O3 -xHost -no-prec-div" FOPTFLAGS="-O3 -xHost -no-prec-div" \
    PETSC_DIR=$(pwd)
''
which  works in my case. Make sure to install some reliable packages such as gcc, gfortran,  make, cmake, and mpich, etc., before installing PETSC.  I think install  PETSC is not easy, sometimes we may meet software incompatibility or system environment,  and need to try and find solutions. Once you have insatll petsc, you can install the solver which according to  your specific needed. 

best regards

**Comment by 123fgh123** [comment]
Thank you very much and I will have a try.

Kind regards

**Reply by MarDiehl** [reply]
@123fgh123: We indeed don't provide details regarding the PETSc installation because it has a very good documentation and the amazing PETSc team answers all your questions faster and quicker than we could do.
